zap/src/test/java/org/zaproxy/zap/authentication/AuthenticationMethodPollUrlUnitTest.java

```diff
@@ -279,4 +279,62 @@ void shouldPollOnSpecifiedNumberOfRequestsPerUser() throws NullPointerException,
         // Second poll for user2
         assertThat(orderedReqs.size(), is(4));
     }
+
+    @Test
+    void shouldHandlePollHeadersWithColonsInValues() throws Exception {
+        // Given
+        String test = "/shouldHandlePollHeadersWithColonsInValues/test";
+        String pollUrl = "/shouldHandlePollHeadersWithColonsInValues/pollUrl";
+        final List<String> orderedReqs = new ArrayList<>();
+        final List<HttpMessage> capturedPollMessages = new ArrayList<>();
+
+        setMessageHandler(
+                msg -> {
+                    String path = msg.getRequestHeader().getURI().getPath();
+                    if (pollUrl.equals(path)) {
```

Comment:
There's no need to check the path, we are after the headers, also if we wanted to check the path we could do so with the message that was sent.